data-prepper-core/src/test/java/org/opensearch/dataprepper/peerforwarder/PeerForwardingProcessingDecoratorTest.java

```diff
@@ -16,54 +18,102 @@
 
 import java.util.Collection;
 import java.util.Collections;
+import java.util.List;
+import java.util.Set;
 
 import static org.mockito.ArgumentMatchers.any;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.CoreMatchers.equalTo;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoInteractions;
+import static org.mockito.Mockito.when;
 
 @ExtendWith(MockitoExtension.class)
 class PeerForwardingProcessingDecoratorTest {
+    private static final String TEST_PLUGIN_ID = "test_plugin_id";
+    private static final String TEST_IDENTIFICATION_KEY = "identification_key";
+
     private Record<Event> record;
 
     @Mock
-    Processor processorMock;
+    Processor processor;
+
+    @Mock(extraInterfaces = Processor.class)
```

Comment:
Hey folks (@dlvenable, @chenqi0805 , @asifsmohammed), was the intention to have [RequiresPeerForwarding](https://github.com/opensearch-project/data-prepper/blob/31bf29b5c562ed74cecd65760c38c0a91edbca00/data-prepper-api/src/main/java/com/amazon/dataprepper/model/peerforwarder/RequiresPeerForwarding.java#L14) be its own interface? Do we plan on having other plugins implement RequiresPeerForwarding? Would we want the interface to extend the [Processor interface](https://github.com/opensearch-project/data-prepper/blob/main/data-prepper-api/src/main/java/com/amazon/dataprepper/model/processor/Processor.java)?

Reply:
My understanding is yes. Plugins(Processors) that needs peer-forwarding will implement RequiresPeerForwarding, e.g. service-map-stateful, aggregate, they have already extended the Proccessor interface. But I suppose that is close to the last step.